src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -2021,7 +2026,15 @@ private static function createRequestFromFactory(array $query = array(), array $
      */
     public function isFromTrustedProxy()
     {
-        return self::$trustedProxies && IpUtils::checkIp($this->server->get('REMOTE_ADDR'), self::$trustedProxies);
+        if (!self::$trustedProxies) {
+            return false;
+        }
+
+        if (!isset($this->validTrustedProxyIP)) {
```

Comment:
It is possible that `self::$trustedProxies` is changed between method calls. Make sure to reset the cached result in `setTrustedProxies` and that this is tested.